virttest/env_process.py

```diff
@@ -1011,15 +1011,21 @@ def preprocess(test, params, env):
             test.cancel("Failed to map hostname and ipaddress of target host")
         session.close()
 
-    # Destroy and remove VMs that are no longer needed in the environment
+    # Destroy and remove VMs that are no longer needed in the environment or
+    # leave them untouched if they have to be disregarded only for this test
     requested_vms = params.objects("vms")
+    vm_disposal = params.get_boolean("unrequested_vms_disposal", "yes")
```

Comment:
By the current implementation you can safely skip the whole loop in case `vm_disposal` is True, right?

Anyway do you think it'd be useful to allow per-vm granularity by using `object_params(vm.name)` to get the `unrequested_vms_disposal`? (in such case it'd have to be called `unrequested_vm_disposal`)

Also the param is fairly cryptical, how about `keep_env_vms` or `skip_env_vms_postprocess`?

And one more thing, wouldn't this clash with the https://github.com/avocado-framework/avocado-vt/pull/2322 (I mean it would, but can be solved by the same logic...)

Reply:
> By the current implementation you can safely skip the whole loop in case `vm_disposal` is True, right?

I think you mean "if `vm_disposal` is False" and the answer is not quite because I would still like to (debug) log what vms have been discarded.

> Anyway do you think it'd be useful to allow per-vm granularity by using `object_params(vm.name)` to get the `unrequested_vms_disposal`? (in such case it'd have to be called `unrequested_vm_disposal`)

At least not in out use cases - so far we either keep or we destroy unregistered vms. If this feature proves itself to others (and so far this seems to be a hard task from the reviews) it is not trouble to extend it. But vm granularity is a use case that has to prove itself needed considering people still have to cross the first stage and keep vms overall.

> Also the param is fairly cryptical, how about `keep_env_vms` or `skip_env_vms_postprocess`?

It was proposed by @luckyh earlier today in the comments ending with https://github.com/avocado-framework/avocado-vt/pull/2105#discussion_r366833979 where I agreed with him that using 'env' in the parameter name could be misleading. Maybe we could change to `keep_env_vms` if he agrees and we reach one joint choice.

> And one more thing, wouldn't this clash with the #2322 (I mean it would, but can be solved by the same logic...)

Not sure but this PR is much older (June 2019) and @xutian approved it as well so I guess he is aware of the changes undertaken here.

Reply:
I took the liberty to rename the parameter to `keep_env_vms` without further feedback. Hopefully this will prevent some unnecessary additional back and forth.

Reply:
> It was proposed by @luckyh earlier today in the comments ending with [#2105 (comment)](https://github.com/avocado-framework/avocado-vt/pull/2105#discussion_r366833979) where I agreed with him that using 'env' in the parameter name could be misleading. Maybe we could change to `keep_env_vms` if he agrees and we reach one joint choice.

@pevogam @ldoktor ok, let me clarify my point: it would be better to make the new param decouple with both the concept and the implement of 'env', that's because even without them we can also have this feature (will tell more about this later). The thing we should highlight is those relevant VMs are "unrequested" (or in other words "unregistered") so I put it into the name. And about the last word "disposal" I just want it to express that how we deal with the VMs, as I'm not a native speaker, maybe it's not that good.

With this feature, users can let the VMs they don't requested be alive across tests or even jobs, and no matter whether the maintainers agree about the proposal of job-level persistence the feature still makes sense, so that's why I suggested a separate commit.

Anyway, the name `keep_env_vms` looks good to me too, feel free to pick that, thanks.